src/devices/casparCG.ts

```diff
@@ -897,4 +909,203 @@ export class CasparCGDevice extends DeviceWithState<State> implements IDevice {
+								this._commandReceiver(this.getCurrentTime(), c, 'Internal transition', 'internalTransition')
+								.catch((e => this.emit('error', 'CasparCG.InternalTransition', e)))
+							}
+						)
+						// Abort: don't send the original command
+						return undefined
+					} else {
+						this._transitionHandler.snapTransition(
+							this._getTransitionId('FILL', command.channel, command.layer),
+							[objectParams.x, objectParams.y, objectParams.xScale, objectParams.yScale]
+						)
+					}
+				} else if (objectParams.keyword === 'PERSPECTIVE') {
+					if (objectParams.transition === Transition.TSR_TRANSITION) {
+						// Handle transitions internally:
+
+						this._transitionHandler.activateTransition(
+							this._getTransitionId('PERSPECTIVE', command.channel, command.layer),
+							[0, 0, 1, 0, 1, 1, 0, 1],
+							[
+								objectParams.topLeftX,
+								objectParams.topLeftY,
+								objectParams.topRightX,
+								objectParams.topRightY,
+								objectParams.bottomRightX,
+								objectParams.bottomRightY,
+								objectParams.bottomLeftX,
+								objectParams.bottomLeftY
+							],
+							['tl', 'tl', 'tr', 'tr', 'br', 'br', 'bl', 'bl'],
+							transitionOptions,
+							{
+								'tl': { type: 'physical' }, // top left corner
+								'tr': { type: 'physical' }, // top right corner
+								'bl': { type: 'physical' }, // bottom right corner
+								'br': { type: 'physical' }  // bottom left corner
+							},
+							(newValues) => {
+								const c = new AMCP.MixerPerspectiveCommand({
+									channel: command.channel,
+									layer: command.layer,
+									topLeftX: newValues[0],
+									topLeftY: newValues[1],
+									topRightX: newValues[2],
+									topRightY: newValues[3],
+									bottomRightX: newValues[4],
+									bottomRightY: newValues[5],
+									bottomLeftX: newValues[6],
+									bottomLeftY: newValues[7]
+								})
+								this._commandReceiver(this.getCurrentTime(), c, 'Internal transition', 'internalTransition')
+								.catch((e => this.emit('error', 'CasparCG.InternalTransition', e)))
+							}
+						)
+						// Abort: don't send the original command
+						return undefined
+					} else {
+						this._transitionHandler.snapTransition(
+							this._getTransitionId('PERSPECTIVE', command.channel, command.layer),
+							[
+								objectParams.topLeftX,
+								objectParams.topLeftY,
+								objectParams.topRightX,
+								objectParams.topRightY,
+								objectParams.bottomRightX,
+								objectParams.bottomRightY,
+								objectParams.bottomLeftX,
+								objectParams.bottomLeftY
+							]
+						)
+					}
+				} else if (
+					objectParams.keyword === 'OPACITY' ||
+					objectParams.keyword === 'VOLUME'
+				) {
+					const opt: { initial: number, prop: string } = (
+						objectParams.keyword === 'OPACITY' ?
+						{
+							initial: 1,
+							prop: 'opacity'
+						} :
+						objectParams.keyword === 'VOLUME' ?
+						{
+							initial: 1,
+							prop: 'opacity'
```

Comment:
Given the keyword I would assume this should be `volume`?

Reply:
Congratulations! you found my test-if-the-reviewer-reads-all-of-the-code bit ;)